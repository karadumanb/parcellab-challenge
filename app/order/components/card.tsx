import { Card } from '@/components/ui/card'
import { cn } from '@/lib/utils'
import { ComponentProps } from 'react'

type Props = ComponentProps<typeof Card>

export function OrderCard({ className, ...others }: Props) {
  return (
    <Card
      className={cn('rounded-lg shadow-lg h-full lg:h-3/4 overflow-auto', className)}
      {...others}
    />
  )
}
