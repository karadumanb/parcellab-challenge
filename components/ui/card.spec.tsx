import { render } from '@testing-library/react'
import { Card, CardContent, CardDescription, CardFooter, CardHeader, CardTitle } from './card'

describe('Card', () => {
  it('snapshot tests', () => {
    const { asFragment } = render(
      <Card>
        <CardHeader>
          Header
          <CardTitle>Title</CardTitle>
          <CardDescription>Description</CardDescription>
        </CardHeader>
        <CardContent>Content</CardContent>
        <CardFooter>Footer</CardFooter>
      </Card>
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
