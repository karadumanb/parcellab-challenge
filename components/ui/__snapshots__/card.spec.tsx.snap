// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card snapshot tests 1`] = `
<DocumentFragment>
  <div
    class="rounded-lg border bg-card text-card-foreground shadow-sm"
  >
    <div
      class="flex flex-col space-y-1.5 p-6"
    >
      Header
      <h3
        class="text-2xl font-semibold leading-none tracking-tight"
      >
        Title
      </h3>
      <p
        class="text-sm text-muted-foreground"
      >
        Description
      </p>
    </div>
    <div
      class="p-6 pt-0"
    >
      Content
    </div>
    <div
      class="flex items-center p-6 pt-0"
    >
      Footer
    </div>
  </div>
</DocumentFragment>
`;
